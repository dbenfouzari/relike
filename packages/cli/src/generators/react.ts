import { paramCase, snakeCase } from "change-case";
import fs from "fs";
import path from "path";

import { replace, replaceFileName } from "../utils/template";

/**
 * Generates component index file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateIndexFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateIndexFile = (key: string, componentName: string) => {
  const templateIndexPath = path.resolve(__dirname, "../templates/index/index.ts.txt");

  const templateIndex = fs.readFileSync(templateIndexPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateIndexPath, key, componentName)),
    content: replace(templateIndex, key, componentName, paramCase),
  };
};

/**
 * Generates component style file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateStyleFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateStyleFile = (key: string, componentName: string) => {
  const templateStylePath = path.resolve(__dirname, "../templates/react-style/__name__.module.scss.txt");

  const templateStyle = fs.readFileSync(templateStylePath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateStylePath, key, componentName)),
    content: replace(templateStyle, key, componentName, snakeCase),
  };
};

/**
 * Generates component story file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateStoryFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateStoryFile = (key: string, componentName: string) => {
  const templateStoryPath = path.resolve(__dirname, "../templates/react-story/__name__.stories.tsx.txt");

  const templateStory = fs.readFileSync(templateStoryPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateStoryPath, key, componentName)),
    content: replace(templateStory, key, componentName),
  };
};

/**
 * Generates component test file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateTestFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateTestFile = (key: string, componentName: string) => {
  const templateTestPath = path.resolve(__dirname, "../templates/react-test/__name__.test.tsx.txt");

  const templateTest = fs.readFileSync(templateTestPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateTestPath, key, componentName)),
    content: replace(templateTest, key, componentName),
  };
};

/**
 * Generates component file
 *
 * @param   {string}  key           The key to replace
 * @param   {string}  componentName The component name
 * @param   {boolean} hasStyle      Does the component have style
 * @example
 * generateComponentFile("name", name, flags.style);
 * @returns {any}                   File path and content.
 */
export const generateComponentFile = (key: string, componentName: string, hasStyle: boolean) => {
  const templateComponentPath = hasStyle
    ? path.resolve(__dirname, "../templates/react-component/__name__(if:style).tsx.txt")
    : path.resolve(__dirname, "../templates/react-component/__name__.tsx.txt");

  const templateComponent = fs.readFileSync(templateComponentPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateComponentPath, key, componentName)),
    content: replace(templateComponent, key, componentName),
  };
};
