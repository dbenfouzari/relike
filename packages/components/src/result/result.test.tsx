import { render, screen } from "@testing-library/react";

import Result, { ResultStatus } from "./result";

describe("Result", () => {
  it("should render successfully with success", () => {
    render(<Result title="Hello Testing World!" status={ResultStatus.SUCCESS} />);

    expect(screen.getByTestId("result")).toMatchSnapshot();
  });

  it("should render successfully with info", () => {
    render(<Result title="Your operation has been executed." status={ResultStatus.INFO} />);

    expect(screen.getByTestId("result")).toMatchSnapshot();
  });

  it("should render successfully with warning", () => {
    render(<Result title="There are some problems with your operation." status={ResultStatus.WARNING} />);

    expect(screen.getByTestId("result")).toMatchSnapshot();
  });

  it("should render successfully with error", () => {
    render(
      <Result
        title="Submission Failed"
        subtitle="Please check and modify the following information before resubmitting."
        status={ResultStatus.ERROR}
      />,
    );

    expect(screen.getByTestId("result")).toMatchSnapshot();
  });

  it("should render successfully with unknown status", () => {
    render(
      <Result
        title="Submission Failed"
        subtitle="Please check and modify the following information before resubmitting."
        // @ts-expect-error Because this status does not exist
        status="unknown"
      />,
    );
  });
});
