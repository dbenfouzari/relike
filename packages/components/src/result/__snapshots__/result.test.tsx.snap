// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should render successfully with error 1`] = `
<div
  class="wrapper"
  data-testid="result"
>
  <div
    class="icon"
  >
    <div
      class="icon_wrapper icon_wrapper__error"
      style="width: 72px; height: 72px;"
    >
      <span
        class="jsx-7413529 icon material-icons"
      >
        close
      </span>
    </div>
  </div>
  <div
    class="title"
  >
    Submission Failed
  </div>
  <div
    class="subtitle"
  >
    Please check and modify the following information before resubmitting.
  </div>
</div>
`;

exports[`Result should render successfully with info 1`] = `
<div
  class="wrapper"
  data-testid="result"
>
  <div
    class="icon"
  >
    <span
      class="jsx-467294570 icon material-icons"
    >
      error
    </span>
  </div>
  <div
    class="title"
  >
    Your operation has been executed.
  </div>
</div>
`;

exports[`Result should render successfully with success 1`] = `
<div
  class="wrapper"
  data-testid="result"
>
  <div
    class="icon"
  >
    <div
      class="icon_wrapper icon_wrapper__success"
      style="width: 72px; height: 72px;"
    >
      <span
        class="jsx-7413529 icon material-icons"
      >
        check
      </span>
    </div>
  </div>
  <div
    class="title"
  >
    Hello Testing World!
  </div>
</div>
`;

exports[`Result should render successfully with warning 1`] = `
<div
  class="wrapper"
  data-testid="result"
>
  <div
    class="icon"
  >
    <span
      class="jsx-3637481999 icon material-icons-round"
    >
      warning
    </span>
  </div>
  <div
    class="title"
  >
    There are some problems with your operation.
  </div>
</div>
`;
